rfcs/41-ui-navigation.md

```diff
@@ -0,0 +1,364 @@
+
+## Terminology
+
+- _menu tree_: the access hierarchy of menus.
+- _activating_ a focusable: Sending the `Action` event while this focusable is focused.
+- _focused menu_: The menu in which the currently focused `Focusable` is.
+- _child menu_ (of a focusable): The menu accessed from this focusable.
+
+## Motivation
+
+`bevy_ui` interaction story is crude.
+The only way to interact with the UI currently
+is through the `Interaction` and `FocusPolicy` components.
+`Interaction` only supports mouse or touch-based input.
+We want gamepad supports and more game-oriented functionalities,
+such as menu trees and movement between menus.
+
+The current UI focus system is limited to `bevy_ui` and
+can't be swapped out for something the user deems more appropriate.
+
+This is in contradiction with the general philosophy of bevy,
+which is of providing to the user all the tools necessary to create something
+that fits their needs optimally.
+
+This RFC proposes to resolve all those questions.
+[`bevy-ui-navigation`] provides a highly customizable ECS-based navigation engine.
+
+## User-facing explanation
+
+The [bevy-ui-navigation README][`bevy-ui-navigation`] is a good start.
+To summarize, here are the headlines:
+- The navigation system can only be interacted with with events:
+  `NavRequest` as input and `NavEvent` as output.
+- `NavEvent::FocusChanged` contains the list of entities that are traversed
+  to go from the last focus position to the new one.
+- ui-navigation provides default input systems, 
+  but it's possible to disable them and instead use custom ones.
+  This is why we restrict interactions with navigation to events.
+- ui-navigation provides a system to declare buttons as "leading to menu X"
+  in order to create menu trees.
+- ui-navigation defines [custom `SystemParam`][event_helpers]s
+  to ease reaction to events (such a button press or menu change).
+  The most naive usage results into [elm architecture][elm_architecture]-like code.
+  Systems updating the model (ECS hierarchy) and systems reacting to events. 
+- It is possible to create isolated menus by using the `NavMenu` component.
+- All `Focusable` children in the `NavMenu` entity's tree will be part
+  of this menu, and moving using a gamepad from one focusable in a menu
+  can only lead to another focusable within the same menu.
+  You can specify a focusable in any other menu that will lead to this
+  one once activated.
+
+## Implementation
+
+Since `bevy-ui-navigation` already exists,
+we will discuss its design, it's functionalities and their uses.
+
+Implementation details are already covered by the implementation,
+which btw is fully documented (including internal architecture).
+
+### Exposed API
+
+The crux of `bevy-ui-navigation` are the following types:
+- `Focusable` component
+  - `lock`: activating it will lock the navigation engine and discard all `NavRequest`.
+  - `cancel`: activating it will do the equivalent of receiving a `Cancel` request.
```

Comment:
This feels circular and is very unclear to me.

Reply:
I've replaced this whole section with links to their corresponding rust doc items, since it's pointless to try to summarize the whole library into a few lines.